server/datastore/mysql/scripts.go

```diff
@@ -299,21 +301,17 @@ func (ds *Datastore) listUpcomingHostScriptExecutions(ctx context.Context, hostI
 			IF(ua.activated_at IS NULL, 0, 1) AS topmost
 		FROM
 			upcoming_activities ua
-			INNER JOIN script_upcoming_activities sua
+			-- left join because software_uninstall has no script join
+			LEFT JOIN script_upcoming_activities sua
 				ON ua.id = sua.upcoming_activity_id
 		WHERE
 			ua.host_id = ? AND
-			ua.activity_type = 'script' AND
-			(
-				ua.payload->'$.sync_request' = 0 OR
-				ua.created_at >= DATE_SUB(NOW(), INTERVAL ? SECOND)
```

Comment:
Sync scripts are treated as any other script in the upcoming queue: https://github.com/fleetdm/fleet/issues/22866#issuecomment-2575961141